airflow/www/views.py

```diff
@@ -994,6 +1003,13 @@ def trigger(self, session=None):
             return redirect(origin)
 
         run_conf = {}
+        conf = request.values.get('conf')
+        if conf:
+            try:
+                run_conf = json.loads(conf)
+            except json.decoder.JSONDecodeError:
+                flash("Invalid JSON configuration")
+                return redirect(url_for('Airflow.trigger', dag_id=dag_id))
```

Comment:
Rather than redirecting it would be better to just display the form again -- that input will be re-shown, and origin param preserved etc.